From a84de1fa13c7ac7e9686758e255eb84e9e1f3ec9 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Wed, 18 Feb 2015 11:07:17 +0000
Subject: [PATCH 44/N] Leave paths containing '/~' alone

We do not perform tilde expansion in the MSys2 runtime; let's leave
those paths intact for programs that want to expand such paths
themselves.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/msys2_path_conv.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/winsup/cygwin/msys2_path_conv.cc b/winsup/cygwin/msys2_path_conv.cc
index d9ec504..bdde9d4 100644
--- a/winsup/cygwin/msys2_path_conv.cc
+++ b/winsup/cygwin/msys2_path_conv.cc
@@ -364,6 +364,10 @@ skip_p2w:
         case '[':
         case ']':
             goto skip_p2w;
+        case '/':
+            if (it + 1 < end && it[1] == '~')
+                goto skip_p2w;
+            break;
         ++it;
     }
     it = *src;
